src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,987 @@ static ColliderJntSphInit D_80AD2790 = {
+};
+
+void EnPeehat_SetupAction(EnPeehat* this, EnPeehatActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnPeehat_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    Actor_SetScale(thisx, 36.0f * 0.001f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06001C80, &D_060009C4, this->jointTable, this->morphTable, 24);
+    ActorShape_Init(&thisx->shape, 100.0f, (void*)&ActorShadow_DrawCircle, 27.0f);
+    thisx->focus.pos = thisx->world.pos;
+    this->unk2D4 = 0;
+    thisx->world.rot.y = 0;
+    thisx->colChkInfo.mass = MASS_HEAVY;
+    thisx->colChkInfo.health = 6;
+    thisx->colChkInfo.damageTable = &sDamageTable;
+    thisx->floorHeight = thisx->world.pos.y;
+    Collider_InitCylinder(globalCtx, &this->colCylinder);
+    Collider_SetCylinder(globalCtx, &this->colCylinder, thisx, &sCylinderInit);
+    Collider_InitQuad(globalCtx, &this->colQuad);
+    Collider_SetQuad(globalCtx, &this->colQuad, thisx, &sQuadInit);
+    Collider_InitJntSph(globalCtx, &this->colJntSph);
+    Collider_SetJntSph(globalCtx, &this->colJntSph, thisx, &sJntSphInit, this->colJntSphItemList);
+
+    thisx->naviEnemyId = 0x48;
+    this->xzDistToRise = 740.0f;
+    this->xzDistMax = 1200.0f;
+    thisx->uncullZoneForward = 4000.0f;
+    thisx->uncullZoneScale = 800.0f;
+    thisx->uncullZoneDownward = 1800.0f;
+    switch (thisx->params) {
+        case TYPE_GROUNDED:
+            EnPeehat_Ground_SetStateGround(this);
+            break;
+        case TYPE_FLYING:
+            thisx->uncullZoneForward = 4200.0f;
+            this->xzDistToRise = 2800.0f;
+            this->xzDistMax = 1400.0f;
+            EnPeehat_Flying_SetStateGround(this);
+            thisx->flags &= -2;
+            break;
+        case TYPE_LARVA:
+            thisx->scale.x = thisx->scale.z = 0.006f;
+            thisx->scale.y = 0.003f;
+            this->colCylinder.dim.radius = 25;
+            this->colCylinder.dim.height = 15;
+            this->colCylinder.dim.yShift = -5;
+            this->colCylinder.info.bumper.dmgFlags = 0x1F824;
+            this->colQuad.base.atFlags = AT_ON | AT_TYPE_ENEMY;
+            this->colQuad.base.acFlags = AC_ON | AC_TYPE_PLAYER;
+            thisx->naviEnemyId = 0x49; // Larva
+            EnPeehat_Larva_SetStateSeekPlayer(this);
+    }
+}
+
+void EnPeehat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPeehat* this = THIS;
+    EnPeehat* parent;
+
+    Collider_DestroyCylinder(globalCtx, &this->colCylinder);
+    Collider_DestroyJntSph(globalCtx, &this->colJntSph);
+
+    // If larva, decrement total larva spawned
+    if (thisx->params > 0) {
+        parent = (EnPeehat*)thisx->parent;
+        if (parent != NULL && parent->actor.update != NULL) {
+            parent->unk2FA--;
+        }
+    }
+}
+
+void EnPeehat_SpawnDust(GlobalContext* globalCtx, EnPeehat* this, Vec3f* pos, f32 arg3, s32 arg4, f32 arg5, f32 arg6) {
+    Vec3f pPos;
```

Comment:
Why pPos, pVel, and pAccel and not juse pos, vel, and accel?

Reply:
i named the third argument of that function `pos`, so i had to name the temp var differently. opted for a p prefix for particle, and matched the pattern for the other variables for consistency.

Reply:
id prob go with dustPos dustVel etc